### PR TITLE
Add PRI macros for T_PTR

### PR DESCRIPTION
**Pull request by parth21999**

### CODE REVIEW

**Review by BertKleewein — APPROVED** [review]
:shipit:

File: common/inc/c_pal/thandle_ptr.h [hunk 1]
```diff
@@ -31,6 +31,12 @@ typedef struct PTR_STRUCT_TAG_TYPE_NAME(T)
     THANDLE_PTR_FREE_FUNC_TYPE_NAME(T) dispose; /*original dispose passed to THANDLE_PTR_CREATE_WITH_MOVE*/     \
 } PTR(T);
 
+#define PRI_T_PTR "p"
+
+#define T_PTR_VALUE(t_ptr) ((t_ptr)->pointer)
+
+#define T_PTR_VALUE_OR_NULL(t_ptr) ((t_ptr) == NULL ? NULL : (t_ptr)->pointer)
```
Comment (jebrando) [inline common/inc/c_pal/thandle_ptr.h:38]: Why don't you just combine these and have 1 define 

T_PTR_VALUE(t_ptr) ((t_ptr) == NULL ? NULL : (t_ptr)->pointer)

Comment (jebrando) [reply common/inc/c_pal/thandle_ptr.h:38]: This would make it so it never will crash is you accidently called the wrong one.

**Review by jebrando — APPROVED** [review]
:shipit:

File: common/inc/c_pal/thandle_ptr.h (hunk 1 above)
Comment (parth21999) [reply common/inc/c_pal/thandle_ptr.h:38]: We have different `OR_NULL` print macros elsewhere in the repo. This allows the user to be more specific.

If the user thinks that the value cannot be NULL in a certain situation, then it is better to crash then to continue and get in a weird state.

**Pull request merged**